src/Sulu/Bundle/RouteBundle/PageTree/PageTreeRepository.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\PageTree;
+
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentInspector;
+use Sulu\Bundle\PageBundle\Document\BasePageDocument;
+use Sulu\Bundle\RouteBundle\Content\Type\PageTreeRouteContentType;
+use Sulu\Component\Content\Document\WorkflowStage;
+use Sulu\Component\Content\Metadata\Factory\StructureMetadataFactoryInterface;
+use Sulu\Component\Content\Metadata\PropertyMetadata;
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Sulu\Component\Route\Document\Behavior\RoutableBehavior;
+
+/**
+ * Update the route of documents synchronously.
+ */
+class PageTreeRepository implements PageTreeUpdaterInterface, PageTreeMoverInterface
+{
+    const ROUTE_PROPERTY = 'routePath';
+
+    const TAG_NAME = 'sulu_route.routePath';
```

Comment:
In my opinion, the best place to define them is the `PageTreeRouteContentType`. But it would be a little bit confusing then, because the `PageTreeRouteContentType` doesn't even need these constants. What do you think?